src/hera.cpp

```diff
@@ -50,6 +52,49 @@ struct hera_instance : evm_instance {
 
 namespace {
 
+#if HERA_EVM2WASM_JS
+// NOTE: assumes that pattern doesn't contain any formatting characters (e.g. %)
+string mktemp_string(string pattern) {
+  char tmp[PATH_MAX] = { 0 };
+  size_t len = snprintf(tmp, sizeof(tmp), "%s", pattern.c_str());
+  mktemp(tmp);
+  return string(tmp, len);
+}
+
+vector<uint8_t> evm2wasm(vector<uint8_t> const& input) {
+  string fileEVM = mktemp_string("/tmp/hera.evm2wasm.evm.XXXXXXXX");
+  string fileWASM = mktemp_string("/tmp/hera.evm2wasm.wasm.XXXXXXXX");
+
+  ofstream os;
+  os.open(fileEVM);
+  // print as a hex sting
+  os << hex;
+  for (uint8_t byte: input)
+    os << static_cast<int>(byte);
+  os.close();
+
+  string cmd = string("evm2wasm.js ") + "-e " + fileEVM + " -o " + fileWASM;
+  if (system(cmd.c_str()) != 0) {
+    unlink(fileEVM.c_str());
```

Comment:
If would be nice to refactor it so `unlink()` is in single place.

Reply:
Like?
```
int ret = system(...);
unlink(...);
unlink(...);
if (ret != 0) return..
```

Reply:
Yes.